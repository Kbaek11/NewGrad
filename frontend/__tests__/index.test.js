/* eslint-env jest */

import {shallow} from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import App from '../pages/index.js';

describe('With Enzyme', () => {
    it('App shows "Hello world!"', () => {
        const app = shallow(<App />);

        expect(
            app
                .find('Title')
                .children()
                .text(),
        ).toEqual('Hello World!');
    });
});

describe('With Snapshot Testing', () => {
    it('App shows "Hello world!"', () => {
        const component = renderer.create(<App />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
