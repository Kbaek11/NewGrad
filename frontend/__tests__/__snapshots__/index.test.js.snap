// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello world!" 1`] = `
.c0 {
  color: red;
  font-size: 50px;
}

<div>
  <header>
    <nav>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      |
      <a
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      |
      <a
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
    </nav>
  </header>
  <h1
    className="c0"
  >
    Hello World!
  </h1>
  <footer>
    I am the footer!
  </footer>
</div>
`;
